File: internal/sast/fixtures/javascript/header_negative.js
// internal/sast/fixtures/javascript/header_negative.js
const express = require("express");
const app = express();

app.get("/track", (req, res) => {
  const safe = String(req.query.id || "").replace(/[\r\n]/g, "").slice(0, 64);
  res.setHeader("X-Tracking", safe);
  res.send("ok");
});

File: internal/sast/fixtures/python/header_negative.py
# internal/sast/fixtures/python/header_negative.py
from flask import Flask, request, Response
import re

app = Flask(__name__)

@app.route("/track")
def track():
    raw = request.args.get("id", "")
    safe = re.sub(r"[\r\n]", "", raw)[:64]
    resp = Response("ok")
    resp.headers.set("X-Tracking", safe)
    return resp
